src/main/java/bio/terra/app/controller/DatasetsApiController.java

```diff
@@ -171,22 +177,36 @@ private AuthenticatedUserRequest getAuthenticatedInfo() {
 
   @Override
   public ResponseEntity<EnumerateDatasetModel> enumerateDatasets(
-      @Valid @RequestParam(value = "offset", required = false, defaultValue = "0") Integer offset,
-      @Valid @RequestParam(value = "limit", required = false, defaultValue = "10") Integer limit,
-      @Valid @RequestParam(value = "sort", required = false, defaultValue = "created_date")
-          EnumerateSortByParam sort,
-      @Valid @RequestParam(value = "direction", required = false, defaultValue = "asc")
-          SqlSortDirection direction,
-      @Valid @RequestParam(value = "filter", required = false) String filter,
-      @Valid @RequestParam(value = "region", required = false) String region) {
+      Integer offset,
+      Integer limit,
+      EnumerateSortByParam sort,
+      SqlSortDirection direction,
+      String filter,
+      String region) {
     ControllerUtils.validateEnumerateParams(offset, limit);
-    Set<UUID> resources =
-        iamService
-            .listAuthorizedResources(getAuthenticatedInfo(), IamResourceType.DATASET)
-            .keySet();
-    EnumerateDatasetModel esm =
-        datasetService.enumerate(offset, limit, sort, direction, filter, region, resources);
-    return new ResponseEntity<>(esm, HttpStatus.OK);
+    Map<UUID, Set<IamRole>> idsAndRoles =
+        iamService.listAuthorizedResources(getAuthenticatedInfo(), IamResourceType.DATASET);
+    var datasetEnum =
+        datasetService.enumerate(
+            offset, limit, sort, direction, filter, region, idsAndRoles.keySet());
+
+    List<DatasetSummaryModel> summaries =
+        datasetEnum.getItems().stream().map(DatasetSummary::toModel).collect(Collectors.toList());
+    Map<String, List<String>> roleMap = new HashMap<>();
+    for (DatasetSummary summary : datasetEnum.getItems()) {
```

Comment:
It bugs me that now the controller needs to know about the internal models

Reply:
Not sure what you mean by internal. `DatasetSummary`'s use isn't limited to `DatasetService` or `DatasetDao`. There are calls to `Dataset.getDatasetSummary()` all throughout TDR. I think it makes more sense using this object in the controller rather than having the service layer know about the response objects.

Although I probably won't convince you of this if I haven't already... I can move this into the service layer if you'd prefer it done that way.